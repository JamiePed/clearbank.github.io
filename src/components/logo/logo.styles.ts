import styled, { css } from 'styled-components'
import { colors } from 'src/components/theme'

export const Logo: any = styled.div`
  ${({ width }: any): any => css`
    width: ${width || '100%'};
  `}

  ${({ inverted }: any): any => css`
    path:not(#logo-dot):not(#logo-developer-text) {
      fill: ${inverted ? colors.brandSecondaryDarkest : colors.brandLight};
    }
  `}
`

export const TextPath: any = styled.path``
