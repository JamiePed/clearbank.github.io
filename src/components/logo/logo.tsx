import React from 'react'

import LogoCBDeveloper from 'src/assets/images/logo-cb-developer.inline.svg'
import LogoCB from 'src/assets/images/logo-cb.inline.svg'

import * as Types from './logo.types'
import * as Styles from './logo.styles'

export default ({ inverted, width, showDeveloperLogo }: Types.LogoProps) => {
  return (
    <Styles.Logo width={width} inverted={inverted}>
      {showDeveloperLogo
        ? <LogoCBDeveloper />
        : <LogoCB />}
    </Styles.Logo>
  )
}
