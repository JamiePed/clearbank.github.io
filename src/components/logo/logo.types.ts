export interface LogoProps {
  inverted?: boolean
  width?: string
  showDeveloperLogo?: boolean
}
