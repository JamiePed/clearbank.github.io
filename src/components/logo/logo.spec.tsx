import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './logo'

import * as Types from './logo.types'

let component: ReturnType<typeof render> = null
let fragment: DocumentFragment = null
let root: HTMLElement = null

const defaultProps: Types.LogoProps = {
  inverted: true,
  width: '200px'
}

beforeEach(() => {
  component = render(<Component {...defaultProps} children='children' />)

  fragment = component.asFragment()
  root = component.container.firstChild
})

afterEach(() => {
  fragment = null
  root = null
})

describe('Component', () => {
  test('matches snapshot', () => {
    expect(root).toMatchSnapshot()
  })

  test('has children', () => {
    expect(root.hasChildNodes()).toBeTruthy()
  })

  test('has width attribute from props', () => {
    expect(root).toHaveAttribute('width')
    expect(root.getAttribute('width')).toBe(defaultProps.width)
  })

  test.skip('inverted props changes class', () => {
    component.rerender(<Component inverted={false} width='200px' />)
  })
})
