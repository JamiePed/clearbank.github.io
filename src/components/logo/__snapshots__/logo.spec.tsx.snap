// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component inverted props changes class 1`] = `
.c0 {
  width: 200px;
}

.c0 path:not(#logo-dot):not(#logo-developer-text) {
  fill: #fff;
}

<div
  class="c0"
  width="200px"
>
  <div />
</div>
`;

exports[`Component matches snapshot 1`] = `
.c0 {
  width: 200px;
}

.c0 path:not(#logo-dot):not(#logo-developer-text) {
  fill: #0D326E;
}

<div
  class="c0"
  width="200px"
>
  <div />
</div>
`;
